scripts/scr_planetary_feature/scr_planetary_feature.gml

```diff
@@ -218,6 +218,57 @@ function new_planet_feature(feature_type, other_data={}) constructor{
             variable_struct_set(self, names[i], variable_struct_get(data, names[i]))
         }
 	}
+
+#region serialization
+	
+	serialize = function(index){
+
+		// General data and init
+
+		var	_planetary_feature_data = {
+			_index: index,
```

Comment:
the point of the underscore, is to denote that this is a bookkeeping variable, which should not be touched during save-editing. I'll add comments to make this clearer. I did this way because I'm not sure if the order of elements in an array are preserved during jsonification and dejsonification.